staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -376,6 +379,9 @@ func (wc *watchChan) prepareObjs(e *event) (curObj runtime.Object, oldObj runtim
 func decodeObj(codec runtime.Codec, versioner storage.Versioner, data []byte, rev int64) (runtime.Object, error) {
 	obj, err := runtime.Decode(codec, []byte(data))
 	if err != nil {
+		if fatalOnDecodeError {
+			panic(err) // we are running in a test environment and thus an error here is due to a coder mistake
```

Comment:
Umm this is a too agressive imo, why panic?  Also under what conditions are you wanting to do this?  

/cc @wojtek-t